butane_core/src/db/mod.rs

```diff
@@ -560,28 +560,116 @@ impl ConnectionSpec {
             None => Err(crate::Error::UnknownBackend(self.backend_name.clone())),
         }
     }
+    /// Get the backend name.
+    pub const fn backend_name(&self) -> &String {
+        &self.backend_name
+    }
+    /// Get the database connection string.
+    pub const fn connection_string(&self) -> &String {
+        &self.conn_str
+    }
```

Comment:
Medium term objective is to remove the `pub` on these two fields, but this PR is big enough.